Watchman.DomainModel/Configuration/Services/ConfigurationService.cs

```diff
@@ -0,0 +1,115 @@
+            var serverConfiguration = configurations.GetValueOrDefault(serverId) ?? configurations[DEFAULT_SERVER_ID];
+            return (T)serverConfiguration;
+        }
+
+        public IEnumerable<IMappedConfiguration> GetConfigurationItems(ulong serverId)
+        {
+            return _cachedConfigurationItem.Select(x => x.Value.GetValueOrDefault(serverId) ?? x.Value[DEFAULT_SERVER_ID]);
+        }
+
+        public async Task SaveNewConfiguration(IMappedConfiguration changedConfiguration)
+        {
+            using var session = this._sessionFactory.Create();
+            var existingConfiguration = session.Get<ConfigurationItem>()
+                .FirstOrDefault(x => x.ServerId == changedConfiguration.ServerId && x.Name == changedConfiguration.Name);
+            var baseFormatConfigurationItem = this._configurationMapperService.MapIntoBaseFormat(changedConfiguration);
+            if (existingConfiguration == null)
+            {
+                await session.AddAsync(baseFormatConfigurationItem);
+            }
+            else
+            {
+                existingConfiguration.SetValue(baseFormatConfigurationItem.Value);
+                await session.AddOrUpdateAsync(existingConfiguration);
+            }
+            this.Refresh();
+        }
+
+        public async Task InitDefaultConfigurations()
+        {
+            var configurationsTypes = this._configurationTypesSearcher.ConfigurationTypes;
+            var configurations = configurationsTypes.Select(x =>
+            {
+                var conf = (IMappedConfiguration)Activator.CreateInstance(x, DEFAULT_SERVER_ID);
+                return this._configurationMapperService.MapIntoBaseFormat(conf);
+            });
+            using var session = this._sessionFactory.Create();
+            var existingConfigurations = session.Get<ConfigurationItem>().Where(x => x.ServerId == 0).ToList();
+            foreach (var configuration in configurations.Where(x => existingConfigurations.All(y => x.Name != y.Name)))
+            {
+                await session.AddAsync(configuration);
+            }
+            this.Refresh();
+        }
+
+        public void Refresh()
+        {
+            using var session = this._sessionFactory.Create();
+            var configurationItems = session.Get<ConfigurationItem>().ToList();
+            var mappedConfigurations = this._configurationMapperService.GetMappedConfigurations(configurationItems);
+            if (_configurationVersions != null)
+            {
+                var tasks = new List<Task>();
+                var changedConfigurationItems = configurationItems.Where(x => _configurationVersions.GetValueOrDefault(x.Id) != x.Version);
+                foreach (var changedConfiguration in changedConfigurationItems)
+                {
+                    var sameTypeMappedConfigurations = mappedConfigurations.FirstOrDefault(x => x.Key.Name == changedConfiguration.Name).Value;
```

Comment:
robiąc FirstOrDefault zakładasz że może być null 
a później masz .Value 

albo zrób First() albo sprawdzaj czy to null bo sie wysypie